### PR TITLE
Upgrade to Python 3.11

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
Upgrade to Python 3.11

### How to review
- Tests pass
- Changes make sense

### Checklist
* [ ] eq-translations updated to support any new schema keys which need translation

### CODE REVIEW

**Comment by rmccar** [comment]
[Upgrade to Python 3.11 on non Runner repos (2 SP)](https://trello.com/c/5rkGy42V/5730-upgrade-to-python-311-on-non-runner-repos-2-sp)

**Review by katie-gardner — CHANGES_REQUESTED** [review]
I'm getting warnings that the `poetry.lock` file is not consistent with the toml file, think it needs to be regenerated (it suggests using `--no-update` to avoid all the packages updating at the same time)

**Comment by MebinAbraham** [comment]
> I'm getting warnings that the `poetry.lock` file is not consistent with the toml file, think it needs to be regenerated (it suggests using `--no-update` to avoid all the packages updating at the same time)

It's worth us updating the packing at the same time.

**Review by katie-gardner — APPROVED** [review]
All working post upgrade and matches my generated poetry file 👍

**Review by Yuyuutsu — COMMENTED** [review]
Validator working as expected, and changes look good. I am also getting the 
```
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.
```
if that is an issue

File: pyproject.toml [hunk 1]
```diff
@@ -8,15 +8,15 @@ description = "API for questionnaire schema validation"
...
+python = "^3.11"
 Flask = "^2.0.2"
 jsonpointer = "^2.2"
 structlog = "^21.5.0"
 python-dateutil = "^2.8.2"
 jsonschema = "4.4.0"
 jsonpath-rw = "^1.4.0"
 jsonpath-rw-ext = "^1.2.2"
-eq-translations = {git = "https://github.com/ONSDigital/eq-translations.git", rev = "v4.9.0"}
+eq-translations = {git = "https://github.com/ONSDigital/eq-translations.git", rev = "v4.9.1"}
```
Comment (MebinAbraham) [inline pyproject.toml:19]: While we are here, we should update these versions to the latest available. Useful plugin: https://github.com/MousaZeidBaker/poetry-plugin-up

Comment (MebinAbraham) [reply pyproject.toml]: We should update to latest available, not latest within the version constraints.

Comment (rmccar) [reply pyproject.toml]: Pretty sure thats what I have done, I used `poetry up --latest`

Comment (MebinAbraham) [reply pyproject.toml]: Right, I think I may have been looking at the first commits which wasn't. All good then 👍

Comment (rmccar) [reply pyproject.toml]: Ah yeah, I ran it without `--latest` at first 👍

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**